iceoryx_hoofs/include/iceoryx_hoofs/cxx/string.hpp

```diff
@@ -115,7 +117,7 @@ constexpr TruncateToCapacity_t TruncateToCapacity{};
 /// @brief string implementation with some adjustments in the API, because we are not allowed to throw exceptions or use
 /// heap.
 template <uint64_t Capacity>
-class string
+class string final
```

Comment:
I am torn about this and do not like to define classes as `final` (usually there is no need, and the STL also does not do this).

The idea appears to be to prevent a special case of inheritance that involves polymorphism when the dtor is not virtual. The downside is that it also prevents perfectly fine inheritance where this is not an issue. While we do not derive from this class, we also have no real issue with this kind of problems for multiple reasons. 

The problems require explicitly calling `delete` or the `dtor`, in the automatic case the correct dtor is called for potential derived classes.

TLDR: we can keep this `final` here but I still think the rule is unnecessary and counterproductive for useful designs.

Reply:
@MatthiasKillat Is it not more about saying "Hey, use this class as it is. Please use composition rather than inheritance."? From a pure user-perspective I don't think there are many use-cases to inherit from classes in  `iceoryx_hoofs`.